tests/test_metadata_eq_.py

```diff
@@ -41,34 +37,74 @@ def setUpClass(cls) -> None:
+
+        cls.objects["Metadata"] = Metadata(cls.objects["Timestamp"], {})
+        cls.objects["Signed"] = cls.objects["Timestamp"]
+        cls.objects["Key"] = Key(
+            "id", "rsa", "rsassa-pss-sha256", {"public": "foo"}
+        )
+        cls.objects["Role"] = Role(["keyid1", "keyid2"], 3)
+        cls.objects["MetaFile"] = MetaFile(1, 12, {"sha256": "abc"})
+        cls.objects["DelegatedRole"] = DelegatedRole("a", [], 1, False, ["d"])
+        cls.objects["Delegations"] = Delegations(
+            {"keyid": cls.objects["Key"]}, {"a": cls.objects["DelegatedRole"]}
+        )
+        cls.objects["TargetFile"] = TargetFile(
+            1, {"sha256": "abc"}, "file1.txt"
+        )
+
+    # Keys are class names.
+    # Values are a dictionary. In those dictionaries their keys are class
+    # attributes and values are the new attribute worth.
+    classes_attributes: utils.DataSet = {
+        "Metadata": {"signed": None, "signatures": None},
+        "Signed": {"version": -1, "spec_version": "0.0.0"},
+        "Key": {"keyid": "a", "keytype": "foo", "scheme": "b", "keyval": "b"},
+        "Role": {"keyids": [], "threshold": 10},
+        "Root": {"consistent_snapshot": None, "keys": {}},
+        "MetaFile": {"version": None, "length": None, "hashes": {}},
+        "Timestamp": {"snapshot_meta": None},
+        "Snapshot": {"meta": None},
+        "DelegatedRole": {
+            "name": "",
+            "terminating": None,
+            "paths": [""],
+            "path_hash_prefixes": [""],
+        },
+        "Delegations": {"keys": {}, "roles": {}},
+        "TargetFile": {"length": 0, "hashes": {}, "path": ""},
+        "Targets": {"targets": {}, "delegations": []},
+    }
+
+    @utils.run_sub_tests_with_dataset(classes_attributes)
+    def test_classes_eq_(self, test_case_data: Dict[str, Any]) -> None:
+        obj = self.objects[self.case_name]  # pylint: disable=no-member
 
-    def copy_and_simple_assert(self, obj: Any) -> Any:
         # Assert that obj is not equal to an object from another type
         self.assertNotEqual(obj, "")
-        result_obj = copy.deepcopy(obj)
+        obj_2 = copy.deepcopy(obj)
         # Assert that __eq__ works for equal objects.
-        self.assertEqual(obj, result_obj)
-        return result_obj
+        self.assertEqual(obj, obj_2)
 
-    def test_metadata_eq_(self) -> None:
-        md = Metadata.from_bytes(self.metadata["snapshot"])
-        md_2: Metadata = self.copy_and_simple_assert(md)
-
-        for attr, value in [("signed", None), ("signatures", None)]:
-            setattr(md_2, attr, value)
-            self.assertNotEqual(md, md_2, f"Failed case: {attr}")
+        for attr, value in test_case_data.items():
+            setattr(obj, attr, value)
+            self.assertNotEqual(obj, obj_2, f"Failed case: {attr}")
             # Restore the old value of the attribute.
-            setattr(md_2, attr, getattr(md, attr))
+            setattr(obj_2, attr, getattr(obj, attr))
```

Comment:
Did you mean?
```suggestion
            setattr(obj, attr, getattr(obj_2, attr))
```

Also this does are not actually "restore" the original value but assigns a reference to the attribute in the other object. For the purpose of this test it does not seem to matter. Still, I'd probably make a backup of the original value in the beginning and restore to that.